security/libgcrypt-tests.py

```diff
@@ -51,22 +51,23 @@ def setUp(self):
             git.get_repo(url, destination_dir=self.workdir)
             os.chdir(self.workdir)
             self.srcdir = self.workdir
-            process.run("./autogen.sh")
-            process.run("./configure --enable-maintainer-mode --disable-doc")
-            if build.make(self.workdir):
-                # If the make fails then need to run make with -lpthread
-                output = build.run_make(self.workdir,
-                                        extra_args="CFLAGS+=-lpthread",
-                                        process_kwargs={"ignore_status": True})
-                if output.exit_status:
-                    self.fail("'make' failed.")
         elif run_type == "distro":
             self.srcdir = os.path.join(self.workdir, "libgcrypt-distro")
             if not os.path.exists(self.srcdir):
                 os.makedirs(self.srcdir)
             self.srcdir = smm.get_source("libgcrypt", self.srcdir)
             if not self.srcdir:
                 self.fail("libgcrypt source install failed.")
+            os.chdir(self.srcdir)
+        process.run("./autogen.sh")
+        process.run("./configure --enable-maintainer-mode --disable-doc")
+        if build.make(self.srcdir):
+            # If the make fails then need to run make with -lpthread
+            output = build.run_make(self.srcdir,
+                                    extra_args="CFLAGS+=-lpthread",
+                                    process_kwargs={"ignore_status": True})
```

Comment:
@sacsant as per existing code, for `run_type == distro` I think it is not required to run `./autogen.sh and ./configure`
@nasastry ^^

Reply:
Without these steps the test fails with distro as a value for run_type

 (1/1) security/libgcrypt-tests.py:libgcrypt.test;run-run_type-distro-8228: STARTED
 (1/1) security/libgcrypt-tests.py:libgcrypt.test;run-run_type-distro-8228: FAIL: 'make check' failed. (21.87 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

The log file contains following messages:

[stdlog] 2023-01-01 23:08:23,591 avocado.utils.process DEBUG| [stderr] + cp /root/rpmbuild/SOURCES/curves.c /root/rpmbuild/SOURCES/t-mpi-point.c tests/
[stdlog] 2023-01-01 23:08:23,592 avocado.utils.process DEBUG| [stderr] + exit 0
[stdlog] 2023-01-01 23:08:23,597 avocado.utils.process INFO | Command 'rpmbuild -bp --define '_builddir /home/avocado-fvt-wrapper-ssant-master/results/job-2023-01-01T23.08-0985f9c/test-results/1-security_libgcrypt-tests.py_libgcrypt.test_run-run_type-distro-8228/tmp_diretb8e818/1-security_libgcrypt-tests.py_libgcrypt.test/libgcrypt-distro' /root/rpmbuild/SPECS/libgcrypt.spec' finished with 0 after 0.276987552s
[stdlog] 2023-01-01 23:08:23,597 avocado.utils.process INFO | Running 'make check'
[stdlog] 2023-01-01 23:08:23,599 avocado.utils.process DEBUG| [stderr] make: *** No rule to make target 'check'.  Stop.
[stdlog] 2023-01-01 23:08:23,599 avocado.utils.process INFO | Command 'make check' finished with 2 after 0.001295339s
[stdlog] 2023-01-01 23:08:23,600 avocado.test ERROR|